sdk/core/core-amqp/test/utils/utils.ts

```diff
@@ -0,0 +1,40 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+/**
+ * Similar to Promise.allSettled which isn't available in all of our supported environments.
+ * https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/allSettled
+ */
+export async function settleAllTasks(tasks: Promise<any>[]): Promise<any[]> {
+  const taskList = [...tasks];
+  for (const task of taskList) {
+    task
+      .then(() => {
+        taskList.splice(taskList.indexOf(task), 1);
+        return;
+      })
+      .catch(() => {
+        taskList.splice(taskList.indexOf(task), 1);
+      });
+  }
+
+  // Keep settling each task in the list until they're all gone.
+  while (taskList.length) {
+    try {
+      await Promise.race(taskList);
+    } catch (err) {
+      /* no-op */
+    }
+  }
+
+  const results = [];
+  for (const task of tasks) {
+    try {
+      const result = await task;
+      results.push(result);
+    } catch (err) {
+      results.push(err);
+    }
+  }
+  return results;
```

Comment:
Why would just the above suffice for `settleAllTasks`?

Reply:
*sigh* It would. This is the result of being stuck in the forest. I initially had the first portion so that I can see the order they resolved in, but it turns out depending what node runtime I was running in, I couldn't guarantee the exact same order when using setTimeout(0) for everything, and then I realized it wasn't actually important.

So, yes, I can simplify this immensely.

Reply:
Hey, thats what teammates are for! To help you out of the forest!

Reply:
Forest cut!